sentry-android-core/src/main/java/io/sentry/android/core/SentryPerformanceProvider.java

```diff
@@ -0,0 +1,73 @@
+package io.sentry.android.core;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.pm.ProviderInfo;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.SystemClock;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.DateUtils;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public final class SentryPerformanceProvider extends ContentProvider {
```

Comment:
Performance as a generic name instead of AppStartProvider because we may reuse this in the future for more things like starting the app start transaction etc